modules/labs-react/common/lib/utils/border.ts

```diff
@@ -0,0 +1,258 @@
+const borderInlineEndWidth = (isRTL: boolean) => (isRTL ? 'borderLeftWidth' : 'borderRightWidth');
+
+const borderShorthandProps = {
+  border: 'border',
+  borderTop: 'borderTop',
+  borderRight: 'borderRight',
+  borderBottom: 'borderBottom',
+  borderLeft: 'borderLeft',
+  borderInlineStart,
+  borderInlineEnd,
+};
+
+const borderColors = {
+  borderColor: 'borderColor',
+  borderTopColor: 'borderTopColor',
+  borderRightColor: 'borderRightColor',
+  borderBottomColor: 'borderBottomColor',
+  borderLeftColor: 'borderLeftColor',
+  borderInlineStartColor,
+  borderInlineEndColor,
+};
+
+const borderRadii = {
+  borderRadius: 'borderRadius',
+  borderTopLeftRadius: 'borderTopLeftRadius',
+  borderTopRightRadius: 'borderTopRightRadius',
+  borderBottomLeftRadius: 'borderBottomLeftRadius',
+  borderBottomRightRadius: 'borderBottomRightRadius',
+};
+
+const borderStyles = {
+  borderStyle: 'borderStyle',
+  borderTopStyle: 'borderTopStyle',
+  borderRightStyle: 'borderRightStyle',
+  borderBottomStyle: 'borderBottomStyle',
+  borderLeftStyle: 'borderLeftStyle',
+  borderInlineStartStyle,
+  borderInlineEndStyle,
+};
+
+const borderWidths = {
+  borderWidth: 'borderWidth',
+  borderTopWidth: 'borderTopWidth',
+  borderRightWidth: 'borderRightWidth',
+  borderBottomWidth: 'borderBottomWidth',
+  borderLeftWidth: 'borderLeftWidth',
+  borderInlineStartWidth,
+  borderInlineEndWidth,
+};
+
+/**
+ * A style prop function that takes components props and returns border styles. Some props, such as borderRadius and borderColor, are connected to our design tokens.
+ * If no `BorderStyleProps` are found, it returns an empty object.
+ *
+ * @example
+ * // You'll most likely use `border` with low-level, styled components
+ * const BoxExample = () => (
+ *   <Box border={`solid 1px #333333 ${colors.blackPepper400}`}>Hello, border styles!</Box>
+ * );
+ *
+ */
+export function border<P extends BorderStyleProps & {theme?: PartialEmotionCanvasTheme}>(props: P) {
+  // border will always be used within the context of a component, but eslint doesn't know that
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const {canvas} = useTheme(props.theme);
+  const isRTL = canvas.direction === ContentDirection.RTL;
+  const styles = {};
+  for (const key in props) {
+    if (props.hasOwnProperty(key)) {
+      if (key in borderShorthandProps) {
+        const value = props[key as keyof BorderShorthandStyleProps];
+        let attr: string;
+        if (key === 'borderInlineStart') {
+          attr = borderInlineStart(isRTL);
+        } else if (key === 'borderInlineEnd') {
+          attr = borderInlineEnd(isRTL);
+        } else {
+          attr = borderShorthandProps[key as keyof BorderShorthandStyleProps];
+        }
+        // @ts-ignore TS doesn't like adding a potentially unknown key to an object, but because we own this object, it's fine.
+        styles[attr] = value;
+      }
+
+      if (key in borderColors) {
```

Comment:
Nit: Shouldn't these be `else if` (or include a `continue`)?

Reply:
`if else` would be better here, yeah. Good call 👍🏻